proxy/nginx/lualib/router.lua

```diff
@@ -30,12 +31,21 @@ local key = prefix .. ":routes:" .. host
 
 -- Try to get target for host
 res, err = red:get(key)
+
+-- Exit if route could not be read
+if err then
+    ngx.log(ngx.ERR, "error reading route: " .. err)
+    ngx.exit(ngx.HTTP_INTERNAL_SERVER_ERROR)
+end
+
 if not res or res == ngx.null then
+    ngx.log(ngx.WARN, "no route for host " .. host)
+
     -- Construct Redis key for $wildcard
     key = prefix .. ":routes:$wildcard"
     res, err = red:get(key)
     if not res or res == ngx.null then
-        return
+        ngx.exit(ngx.HTTP_NOT_FOUND)
```

Comment:
I'm not really sure we should exit if a random route fails, could you please expand on this one?

Reply:
Actually simple return will produce error of inability to resolve "fallback" host, because container_url is set to that value by default.

That actually will be handled by @fallbacvk location.

In my copy I do "return ngx.exit(ngx.HTTP_BAD_GATEWAY)".

That forces nginx to @fallback route without trying passing to invalid proxy.